src/time_val.rs

```diff
@@ -27,6 +27,6 @@ impl<'a> ToGlibPtrMut<'a, *mut glib_sys::GTimeVal> for TimeVal {
 
 impl Uninitialized for TimeVal {
     unsafe fn uninitialized() -> TimeVal {
-        mem::zeroed()
+        mem::MaybeUninit::zeroed().assume_init()
```

Comment:
As we want to get rid of this trait anyway, just leave it as `mem::zeroed()`. What you write is exactly equivalent to it apart from being more keypresses.